### DISCUSSION TITLE
Read/Write model files in .opb format

### DISCUSSION THREAD

**Discussion by AlbertZhangHIT** [coin-or/python-mip#286] [Q&A] [answered]
Hi there, I have many model files in .opb format. How could I translate these files into .mps/.lp format so that python-mip could solve it ? Moreover, is there a way to translate the .mps/.lp format model file into .opb format?

**Comment by sebheger** [comment, marked as answer]
Hi @AlbertZhangHIT,

I have never heard about .opd file extension before. .opd files are not supported by python-mip. But gurobi supports it. With a pip installed gurobipy you could read the .opd and then write it as .lp/.mps without solving it.

**Reply by AlbertZhangHIT** [reply]
Thanks a lot, it really solves my problem!

**Reply by AlbertZhangHIT** [reply]
Hi @sebheger ,

I got a really weird problem when using gurobipy. 
I saved my model which has `N` variables into `.opb` format file with the help of gurobipy. But when I loaded the model back from the `.opb` format file using gurobipy, I found that the number of variables changed to `N-1`. 
It is really weird.

Update:
I have solved this problem. There may be some bug in writing out .opb model in gurobipy. It always write variable `x1000` as
 `x100`.
